### ISSUE TITLE
Thanks! TM-T88V working 

### ISSUE THREAD

**Issue by lgturrez** [plinth666/epsonsimplecups#4]
I was looking for a suitable driver for my TM-T88V  and finally I found yours! 
Just tested a bit and I saw that it  always feeds the full Page, instead of stopping after content finishes, but it's not such a problem. 
I Will set links to this Page in the forums I visited, in order to help people with the same problem. 
Thanks again!

**Comment by ChrisVonFrosta** [comment]
I could just agree with you! THANKS A LOT! It works very fine for me.

Nevertheless there are two "issues" that me effected.
The  first is the same you told about. Print size is equal to dokument size allways.
The second "issue" is the delay from starting a printing job until the printer feeds the paper and start print. I have to open an issue for that maybe.  

Thanks again

**Comment by tjeerdytsma** [comment]
Great work! I got it working with the TM-T88V. 
When printing jpg images there is a 2cm white margin at the left and about 1cm of the image missing on the right.
Any clue on how I can fix this?

**Comment by ksze** [comment]
Does anybody know whether this driver works if the TM-T88V is connected via ethernet?

**Comment by plinth666** [comment]
Interesting question.
Unfortunately I don't have any way to test that configuration. 
I can't see why it wouldn't as the connection should be independent of the driver.
The driver only turns pixels into escape sequences that the printer understands. I don't recall how much I ask the printer for data. If there's anything that would break, it would be that.

-------- Original message --------

**Comment by ksze** [comment]
I see.

So let's say if I connect multiple printers of the exact same model to the same Raspberry Pi by USB. Is there a way to uniquely identify each printer, regardless of the USB port?

**Comment by corycpf** [comment]
Yes, CUPS will assign a unique serial number per printer. I've tested this with 4 connected USB printers to the pi, you can use the serial number to identify the device.

**Comment by ksze** [comment]
@corycpf does the serial number for each printer stick even if you plug the printer to a different USB port?

**Comment by groolot** [comment]
I tried the package from plinth666 but the raster wasn't quite proper and the print timing waas quiet long.

I used then a bunch of free code from Epson, a PPD file from another printer and modify them to produce this new package: https://github.com/groolot/epson-tm-t88v-driver

Better rasterization, cash drawer and buzzer, all is operational.

Hope that could help someone like I was: in pain.

**Comment by lgturrez** [comment]
grooot, I tried your ppd file but file not found in usr/lib/filter error is shown in CUPS admin page :(

**Comment by groolot** [comment]
That's because you have to compile the source code with the autotools toolchain: `./configure && make && make install`. The only PPD file is not enough, you need the raster filter.

Then you'll find in CUPS admin page the correct PPD line wether you've chosen EPSON as vendor ;)
